File: mili-server/src/main/java/fun/bm/mili/config/modules/function/protocol/XaeroMapProtocolConfig.java
package fun.bm.mili.config.modules.function.protocol;

import com.electronwill.nightconfig.core.file.CommentedFileConfig;
import fun.bm.mili.config.modules.ConfigModule;
import me.earthme.luminol.config.flags.ConfigClassInfo;
import me.earthme.luminol.config.flags.ConfigInfo;
import me.earthme.luminol.enums.EnumConfigCategory;

@ConfigClassInfo(category = EnumConfigCategory.FUNCTION, name = "xaeromap", directory = {"protocol"})
public class XaeroMapProtocolConfig implements ConfigModule {
    @ConfigInfo(name = "enabled", comments = "Enable XaeroMap protocol support") public static boolean enabled = false;
    @ConfigInfo(name = "xaero-map-server-id", comments = "XaeroMap server ID") public static int xaeroMapServerID = 0;
    @Override public void onLoaded(CommentedFileConfig c) {}
    @Override public void onUnloaded(CommentedFileConfig c) {}
}

File: mili-server-kotlin/src/main/kotlin/fun/bm/mili/config/modules/function/protocol/XaeroMapProtocolConfig.kt
package `fun`.bm.mili.config.modules.function.protocol
import com.electronwill.nightconfig.core.file.CommentedFileConfig
import `fun`.bm.mili.config.modules.ConfigModule
import me.earthme.luminol.config.flags.ConfigClassInfo
import me.earthme.luminol.config.flags.ConfigInfo
import me.earthme.luminol.enums.EnumConfigCategory

@ConfigClassInfo(category = EnumConfigCategory.FUNCTION, name = "xaeromap", directory = ["protocol"])
class XaeroMapProtocolConfig : ConfigModule {
    companion object {
        @ConfigInfo(name = "enabled", comments = "Enable XaeroMap protocol support")
        @JvmField var enabled = false
        @ConfigInfo(name = "xaero-map-server-id", comments = "XaeroMap server ID") @JvmField var xaeroMapServerID = 0
    }
    override fun onLoaded(c: CommentedFileConfig) {}
    override fun onUnloaded(c: CommentedFileConfig) {}
}